Assignment 26: Add error handling and some extra credits

Phidelux opened the pull request:
This PR introduces various improvements and fixes:

- Fix formatting
- Add missing includes
- Use of getaddrinfo to setup server addrinfo
- Pass port as commandline argument
- Add missing error handling
- Add missing return statement
- Monitor connecting and disconnecting clients
- Accept parallel connections

Phidelux commented:
Hey,

thanks for the quick merge. I really enjoyed looking threw the exercises. However, I forgot updating the documentation of assignment 26. I will add this during the weekend. Should i split up the necessary changes like fixes, error handling and readability improvements and the extensions like parallel connections and client monitoring? I think the later two would make a good 'Extra Credits' section.

Best regards
Andreas

h0mbre commented:
Hey Andreas,

Thank you very much for the pull requests. I really appreciate it, I gave
you a shoutout on twitter. My only concern with the more complex pull
requests for the last few assignments is that I've already written walk
throughs in the READMEs for those last couple of assignments for *my* code
and I don't want to change those too much. Frankly, your code is much
better than mine and I'd love to add your .c files for the last few
assignments as stand alone 'Extra Credits'. I think that's a great idea. So
if you want, you can submit a pull request to add your assignments at the
end as extra credit assignments. Let me know if you need anything else from
me, thanks again for the contributions!

Regards,
h0mbre

On Fri, Feb 7, 2020 at 2:11 AM Andreas Schickedanz <notifications@github.com>
wrote:

> Hey,
>
> thanks for the quick merge. I really enjoyed looking threw the exercises.
> However, I forgot updating the documentation of assignment 26. I will add
> this during the weekend. Should i split up the necessary changes like
> fixes, error handling and readability improvements and the extensions like
> parallel connections and client monitoring? I think the later two would
> make a good 'Extra Credits' section.
>
> Best regards
> Andreas
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/h0mbre/Learning-C/pull/13?email_source=notifications&email_token=ALK7QGV7LYEFRR4B73WKWCTRBUCRXA5CNFSM4KRFCZI2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELB6M3Y#issuecomment-583263855>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALK7QGRQS436F2VO7ASHAYTRBUCRXANCNFSM4KRFCZIQ>
> .
>

Phidelux commented:
Good morning.

I just applied all reasonable patches, like formatting, return value, missing includes, etc., to the original source and added a separate source for the extra credits. I also updated the README to include all instructions for the extra credits at the end of the document. For the references to the needed function calls i used the linux manual pages hosted by [die.net](https://linux.die.net/man/).

Regards
Andreas

Phidelux commented:
Any update on this one?

Phidelux commented:
Ping. :wink:
